internal/gateway/retrieve.go

```diff
@@ -24,12 +24,67 @@ import (
 	"context"
 	"fmt"
 	"strconv"
+	"time"
 
 	metal "github.com/equinix/equinix-sdk-go/services/metalv1"
 
 	"github.com/spf13/cobra"
 )
 
+type ipreservationIsh interface {
+	GetAddress() string
+	GetCidr() int32
+}
+
+var (
+	_ ipreservationIsh = (*metal.VrfIpReservation)(nil)
+	_ ipreservationIsh = (*metal.IPReservation)(nil)
+)
+
+type gatewayIsh interface {
+	GetId() string
+	GetCreatedAt() time.Time
+	GetState() metal.MetalGatewayState
+	GetVirtualNetwork() metal.VirtualNetwork
+	GetIpReservation() ipreservationIsh
+	GatewayType() string
+}
+
+var (
+	_ gatewayIsh = (*metalGatewayWrapper)(nil)
+	_ gatewayIsh = (*vrfMetalGatewayWrapper)(nil)
+)
+
+type metalGatewayWrapper struct {
+	*metal.MetalGateway
+}
+
+func (m metalGatewayWrapper) GetIpReservation() ipreservationIsh {
+	if m.MetalGateway.IpReservation == nil {
+		return nil
+	}
+	return m.MetalGateway.IpReservation
+}
+
+func (m metalGatewayWrapper) GatewayType() string {
+	return "MetalGateway"
```

Comment:
Is this name ok?

Reply:
Other customer-facing tools (web console, terraform, Ansible, etc.) don't expose a type for gateways so I don't think we should introduce one here.  Terraform exposes a `vrf_id` attribute that is taken from the IP reservation, and we could emulate that in the text output for gateway commands.

Reply:
I didn't find examples of other IDs being included in the output, so I included VRF Name. The meant adding "vrf" to the `include` list as only the `href` was available.